pkg/cli/cmd_helpers.go

```diff
@@ -17,45 +17,256 @@ limitations under the License.
 package cli
 
 import (
+	"errors"
 	"fmt"
+	"os"
 
+	"github.com/spf13/afero"
 	"github.com/spf13/cobra"
 
-	"sigs.k8s.io/kubebuilder/v3/pkg/cli/internal/config"
+	"sigs.k8s.io/kubebuilder/v3/pkg/config/store"
+	yamlstore "sigs.k8s.io/kubebuilder/v3/pkg/config/store/yaml"
+	"sigs.k8s.io/kubebuilder/v3/pkg/model/resource"
 	"sigs.k8s.io/kubebuilder/v3/pkg/plugin"
 )
 
+// noResolvedPluginError is returned by subcommands that require a plugin when none was resolved.
+type noResolvedPluginError struct{}
+
+// Error implements error interface.
+func (e noResolvedPluginError) Error() string {
+	return "no resolved plugin, please verify the project version and plugins specified in flags or configuration file"
+}
+
+// noAvailablePluginError is returned by subcommands that require a plugin when none of their specific type was found.
+type noAvailablePluginError struct {
+	subcommand string
```

Comment:
imo we ought to be a warning to allow users that used SDK tool to scaffold the project to still able to use KB tool.

Reply:
This is considered in the follow up #2019 though the `Bundle` concept. Not being able to resolve a plugin definetely needs to be an error or we will make the users get to broken states.